Classes/Update/MigrateSchedulerTasks.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+namespace Causal\IgLdapSsoAuth\Update;
+
+use Causal\IgLdapSsoAuth\Task\ImportUsers;
+use Doctrine\DBAL\Exception as DBALException;
+use RuntimeException;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Install\Updates\DatabaseUpdatedPrerequisite;
+use TYPO3\CMS\Install\Updates\UpgradeWizardInterface;
+use TYPO3\CMS\Scheduler\Task\ExecuteSchedulableCommandTask;
+
+class MigrateSchedulerTasks implements UpgradeWizardInterface
+{
+    protected string $tablename = 'tx_scheduler_task';
+
+    /**
+     * @return string
+     */
+    public function getIdentifier(): string
+    {
+        return self::class;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTitle(): string
+    {
+        return 'ig_ldap_sso_auth: Migrate scheduler tasks into Symfony commands';
+    }
+
+    /**
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'Beware : this script will split scheduler with configuration "all" into a scheduler task per configuration';
+    }
+
+    /**
+     * @return bool
+     */
+    public function updateNecessary(): bool
+    {
+        return !empty($this->getOldSchedulerTasks());
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getPrerequisites(): array
+    {
+        return [
+            DatabaseUpdatedPrerequisite::class
+        ];
+    }
+
+    /**
+     * @return bool
+     */
+    public function executeUpdate(): bool
+    {
+        $oldSchedulerTasks = $this->getOldSchedulerTasks();
+        foreach ($oldSchedulerTasks as $oldSchedulerTask) {
+            /** @var \Causal\IgLdapSsoAuth\Task\ImportUsers $oldTask */
+            $oldTask = unserialize($oldSchedulerTask['serialized_task_object']);
```

Comment:
As discussed IRL, I plan to remove the old class altogether, thus you won't be able to `unserialize()` the legacy scheduler tasks :)